src/shared/tools.cpp

```diff
@@ -4,36 +4,31 @@
 
 ////////////////////////// rnd numbers ////////////////////////////////////////
 
-#define N (624)
-#define M (397)
-#define K (0x9908B0DFU)
+std::random_device rndseed;
```

Comment:
We should provide a short comment why we store our seed statically. Of course it makes sense to do so (otherwise you could just use the random device directly, which is, in itself, a form of RNG). Future readers however might not be into PRNGs as much.